src/main/java/uk/gov/hmcts/reform/civil/model/CaseData.java

```diff
@@ -697,4 +702,56 @@ public YesOrNo getRespondent2Represented() {
             .filter(Objects::nonNull)
             .findFirst().orElse(null);
     }
+
+    @JsonIgnore
+    public DefendantResponseStatus getDefendantResponseStatus() {
+        if (defenceAdmitPartPaymentTimeRouteRequired != null
+            && defenceAdmitPartPaymentTimeRouteRequired != RespondentResponsePartAdmissionPaymentTimeLRspec.IMMEDIATELY) {
+            return DefendantResponseStatus.ELIGIBLE_FOR_CCJ;
+        }
+        if (respondent1ClaimResponsePaymentAdmissionForSpec == RespondentResponseTypeSpecPaidStatus.PAID_FULL_OR_MORE_THAN_CLAIMED_AMOUNT
+            && respondent1DQ.getResponseClaimCourtLocationRequired() != null) {
+            return DefendantResponseStatus.CLAIMANT_ACCEPTED_STATES_PAID;
+        }
+        if (respondent1ClaimResponsePaymentAdmissionForSpec == RespondentResponseTypeSpecPaidStatus.PAID_FULL_OR_MORE_THAN_CLAIMED_AMOUNT) {
+            return DefendantResponseStatus.PAID_IN_FULL;
+        }
+        if (applicant1ProceedWithClaim != null && respondent1DQ.getResponseClaimCourtLocationRequired() != null) {
+            return DefendantResponseStatus.REDETERMINATION_BY_JUDGE;
+        }
+        if (respondToClaim != null
+            && respondToClaim.getWhenWasThisAmountPaid() != null
+            && respondent1DQ != null
+            && respondent1DQ.getResponseClaimCourtLocationRequired() != null
+            && isCCJPaidWithinMonth()) {
+            return DefendantResponseStatus.PAID_IN_FULL_CCJ_CANCELLED;
+        }
+        if (respondToClaim != null
```

Comment:
This if stmt can be merged with the previous one, similar to my previous comment.